public/main.ts

```diff
@@ -132,27 +132,196 @@ const kingdoms = [
+            player.farms.set(crop, new Farm(1));
+            const stats = document.getElementById("gold-grain-stats");
+            const grainStat = document.createElement("p");
+            grainStat.id = Crop[crop] + "-stats";
+            setElementInnerHTML(grainStat, player.farms.get(crop)!.stockpile.toString() + " " + Crop[crop]);
+
+            const farms = document.getElementById("farms");
+            const farmOption = document.createElement("option");
+            farmOption.id = Crop[crop] + "-farm-option";
+            setElementInnerHTML(farmOption, Crop[crop]);
+            farms!.appendChild(farmOption);
 
+            const sellCrops = document.getElementById("crops-sale");
+            const cropOption = document.createElement("option");
+            cropOption.id = Crop[crop] + "-crop-option";
+            setElementInnerHTML(cropOption, Crop[crop]);
+            sellCrops!.appendChild(cropOption);
 
-window.setInterval(() => {
-    timer += tickRate;
+            stats?.appendChild(grainStat);
+        }
+        else player.farms.get(crop)!.stockpile += 1;
 
+        player.gold-= cropMarket.get(crop)!;
+    }
+
+
+}
+
+// load gui
+function loadGUI() {
+    cropMarket.forEach((value, key, map) => {
+        const button = document.createElement("button");
+        button.onclick = handleBuy;
+        button.id = Crop[key] + "-store";
+        console.log(button, key, value);
+        //button.innerHTML = Crop[key] + " " + value.toString() + "G";
+        setElementInnerHTML(button, Crop[key] + " " + value.toString() + "G");
+        const store = document.getElementById("store-tab");
+        store?.appendChild(button);
+    });
+
+    setIdInnerHTML("gold", player.gold.toString());
+
+    player.farms.forEach((value: Farm, key: Crop, map: Map<Crop, Farm>) => {
+        const parent = document.getElementById("gold-grain-stats");
+        const grain = document.createElement("p");
+        grain.id = Crop[key] + "-stats";
+        setElementInnerHTML(grain, value.stockpile + " " + Crop[key]);
+        parent!.appendChild(grain);
+
+        const farms = document.getElementById("farms");
+        const farmOption = document.createElement("option");
+        farmOption.id = Crop[key] + "-farm-option";
+        setElementInnerHTML(farmOption, Crop[key]);
+        farms!.appendChild(farmOption);
+
+        const sellCrops = document.getElementById("crops-sale");
+        const cropOption = document.createElement("option");
+        cropOption.id = Crop[key] + "-crop-option";
+        setElementInnerHTML(cropOption, Crop[key]);
+        sellCrops!.appendChild(cropOption);
+    });
+
+
+
+
+    setIdInnerHTML("people", player.idlePopulation.toString());
+    setIdInnerHTML("soldiers", player.army.totalSoldiers.toString());
+    setIdInnerHTML("farmers", getTotalFarmers().toString());
+}
+function getTotalFarmers() {
+    let totalFarmers = 0;
+    player.farms.forEach((value: Farm, key: Crop, map: Map<Crop, Farm>) => {
```

Comment:
This callback function could be really simplified.
```suggestion
    player.farms.forEach(value => {
```